### PR TITLE
Update HTTP span name extractors

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**
Implements https://github.com/open-telemetry/opentelemetry-specification/issues/2998

### CODE REVIEW

**Comment by mateuszrzeszutek** [comment]
I'm getting some Vaadin failures on CI that I cannot reproduce locally:
```
org.openqa.selenium.ElementClickInterceptedException: element click intercepted: Element <vaadin-button id="main.button" tabindex="0" role="button">...</vaadin-button> is not clickable at point (77, 76). Other element would receive the click: <div class="v-system-error" onclick="this.parentElement.removeChild(this)">...</div>
```
@laurit is it possible that Vaadin tests use the latest possible version of some JS library that was just released today (e.g. [this one](https://github.com/vaadin/web-components/releases))?

**Comment by laurit** [comment]
@mateuszrzeszutek reproduces after `./gradlew :instrumentation:vaadin-14.2:javaagent:clean` and `rm -rf ~/.pnpm-store/` Probably because frontend compilation fails.

File: testing-common/src/main/groovy/io/opentelemetry/instrumentation/test/base/HttpClientTest.groovy [hunk 1]
```diff
@@ -428,7 +428,7 @@ abstract class HttpClientTest<REQUEST> extends InstrumentationSpecification {
   }
 
   protected String expectedClientSpanName(URI uri, String method) {
-    return method != null ? "HTTP " + method : "HTTP request"
+    return method != null ? method : "HTTP request"
```
Comment (trask) [inline testing-common/src/main/groovy/io/opentelemetry/instrumentation/test/base/HttpClientTest.groovy:431]: do we have any tests with null method somehow?
```suggestion
    return method
```

File: testing-common/src/main/java/io/opentelemetry/instrumentation/testing/junit/http/HttpClientTestOptions.java [hunk 2]
```diff
@@ -32,7 +32,7 @@ public abstract class HttpClientTestOptions {
                   SemanticAttributes.HTTP_USER_AGENT)));
 
   public static final BiFunction<URI, String, String> DEFAULT_EXPECTED_CLIENT_SPAN_NAME_MAPPER =
-      (uri, method) -> method != null ? "HTTP " + method : "HTTP request";
+      (uri, method) -> method != null ? method : "HTTP request";
```
Comment (trask) [inline testing-common/src/main/java/io/opentelemetry/instrumentation/testing/junit/http/HttpClientTestOptions.java:35]: (same q)
```suggestion
      (uri, method) -> method;
```

File: instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpSpanNameExtractor.java [hunk 3]
```diff
@@ -33,13 +33,10 @@ private HttpSpanNameExtractor(HttpCommonAttributesGetter<REQUEST, ?> getter) {
...
-    if (route != null) {
-      return route;
-    }
     String method = getter.getMethod(request);
+    String route = extractRoute(request);
     if (method != null) {
-      return "HTTP " + method;
+      return route == null ? method : method + " " + route;
     }
     return "HTTP request";
```
Comment (trask) [inline instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpSpanNameExtractor.java:41]: wdyt of

```suggestion
    return "HTTP";
```

or

```suggestion
    return "UNKNOWN";
```

File: testing-common/src/main/groovy/io/opentelemetry/instrumentation/test/base/HttpClientTest.groovy (hunk 1 above)
Comment (mateuszrzeszutek) [reply testing-common/src/main/groovy/io/opentelemetry/instrumentation/test/base/HttpClientTest.groovy:431]: Nope, I don't think we do; method is provided by every library we instrument.

File: instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpSpanNameExtractor.java (hunk 3 above)
Comment (mateuszrzeszutek) [reply instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpSpanNameExtractor.java]: I think I like `HTTP` more; if nothing else is provided, no attributes at all, at least it says that it was an HTTP call.

File: instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/internal/HttpRouteState.java [hunk 4]
```diff
@@ -24,14 +24,18 @@ public static HttpRouteState fromContextOrNull(Context context) {
     return context.get(KEY);
   }
 
-  public static HttpRouteState create(int updatedBySourceOrder, @Nullable String route) {
-    return new HttpRouteState(updatedBySourceOrder, route);
+  public static HttpRouteState create(
+      @Nullable String method, int updatedBySourceOrder, @Nullable String route) {
```
Comment (trask) [inline instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/internal/HttpRouteState.java:28]: thoughts on changing ordering? this seems more natural to me:

```suggestion
  public static HttpRouteState create(
      @Nullable String method, @Nullable String route, int updatedBySourceOrder) {
```

**Review by trask — APPROVED** [review]

**Review by laurit — APPROVED** [review]

**Pull request merged**